mlir/test/Conversion/GPUToLLVMSPV/gpu-to-llvm-spv.mlir

```diff
@@ -383,7 +308,20 @@ gpu.module @shuffles attributes {
     %shuffleResult4, %valid4 = gpu.shuffle up %f16_val, %offset, %width : f16
     %shuffleResult5, %valid5 = gpu.shuffle up %f32_val, %offset, %width : f32
     %shuffleResult6, %valid6 = gpu.shuffle down %f64_val, %offset, %width : f64
-    return
+    llvm.return
+  }
+}
+
+// -----
+
+// Check attaching a discardable attribute to a `gpu.func` hosting a `gpu.shuffle` operation works,
+// i.e., the attribute is propagated.
+
+gpu.module @shuffles_ {
+  gpu.func @propagated_size(%val: i32, %id: i32) attributes {intel_reqd_sub_group_size = 16 : i32} {
```

Comment:
```suggestion
  gpu.func @propagated_size(%val: i32, %id: i32) attributes {llvm.intel_reqd_sub_group_size = 16 : i32} {
```
I thought discardable attributes needed a prefix

Reply:
This isn't a discardable attribute from the `llvm` dialect, but an attribute that will be propagated in the conversion to `llvm.func` and understood as the `llvm.func` op attribute.

This is a use-case I am not interested in, but an example of how you could have a parent function, convert that to LLVM and then convert the `gpu.shuffle` operation. Do you think it'd be better to drop this case?

Reply:
if it's not your use case then I maybe wouldn't worry about it. Add what you need, right?
I will say that you have called it a discardable attribute in the comment above.
I could be wrong about this but I was under the impression that there are inherent and discardable attributes, and this seems to be neither.